gcloud-java-dns/src/main/java/com/google/gcloud/dns/spi/DefaultDnsRpc.java

```diff
@@ -54,6 +62,26 @@ public DefaultDnsRpc(DnsOptions options) {
   }
 
   @Override
+  public BatchRequest enqueueListZones(BatchRequest batch, final DnsBatch.Request request,
```

Comment:
This is a tricky one. I considered many options and this seems the best one to me. Below is the list. Long story short, something has to be exposed in `DnsRpc` interface and it should make sense, but there is no clear cut here.

1. The cleanest would be to have `DnsRpc` take care of submitting the batch completely so the exposed method would be `submitBatch(DnsBatch toSubmit)`. `DnsRpc` would then take care of everything but this includes paging. A page implementation requires page fetcher which is in the `DnsImpl` class in a different package. So, this option is out.
2. We can do everything from the DnsImpl class. However, this is completely bypassing the Rpc facade as the DnsImpl makes calls on its own. It is not the worst solution, but as a user, I would expect that if I provide my own implementation for `DnsRpc`, all the calls would be handled using that one.
3. My chosen solution was to use `DnsImpl` to create a collection of requests inside the `BatchRequest`. The callbacks are created in `Dns` which has the paging helpers, and the quing is handled via `Rpc` method, which is exposed. The `BatchRequest` object is created inside `Rpc` as it requires `com.google.api.services.dns.Dns;` which already contain the `HttpTransport`. Doing anything more in `DnsImpl` would require defining the `Rpc` functionality in `DnsImpl` and that is again bypassing the `DnsRpc` layer and duplicating the code.

I will welcome more suggestions, but for now, I am not touching this.

/cc @ajkannan @aozarov